provider/win_driver_utils.py

```diff
@@ -194,7 +194,116 @@ def copy_file_to_samepath(session, test, params):
+    :param vm: vm object
+    :param test: kvm test object
+    :param cmd: Driver disable command
+    :param device_id: Device id in windows guest
+    """
+    cmd = cmd.replace("DEVICE_ID", device_id)
+    cmd = utils_misc.set_winutils_letter(session, cmd)
+    status, output = session.cmd_status_output(cmd)
+    if status != 0:
+        if "reboot" in output:
+            vm.reboot()
+        else:
+            test.fail("failed to disable driver, %s" % output)
+
+
+def get_device_id(session, test, driver_name):
+    """
+    Get device id from guest.
+
+    :param session: The guest session object
+    :param test: kvm test object
+    :param driver_name: Driver name
+    """
+    device_name = driver_info_dict[driver_name]["device_name"]
+    device_hwid = driver_info_dict[driver_name]["hwid"]
+
+    output = _pnpdrv_info(session, device_name, ["DeviceID"])
+    # workaround for viostor/vioscsi to get data device id
+    device_id = output[0]
+    device_id = '&'.join(device_id.split('&amp;'))
+    find_devices = False
+    for hwid in device_hwid.split():
+        hwid = hwid.split('"')[1]
+        if hwid in device_id:
+            find_devices = True
+    if not find_devices:
+        test.fail("Didn't find driver info from guest %s" % output)
+    return device_id
+
+
+def load_driver(session, test, params, load_method='enable'):
+    """
+    Load driver.
+
+    :param session: The guest session object
+    :param test: kvm test object
+    :param params: the dict used for parameters
+    :param load_method: Load driver method
+    """
+    driver_name = params["driver_name"]
+    if load_method != 'enable':
+        devcon_path = params["devcon_path"]
+        media_type = params.get("virtio_win_media_type", "iso")
+
+        device_hwid = driver_info_dict[driver_name]["hwid"]
+        return install_driver_by_virtio_media(session, test, devcon_path,
+                                              media_type, driver_name,
+                                              device_hwid)
+    else:
+        cmd = params["driver_enable_cmd"]
+        device_id = get_device_id(session, test, driver_name)
+        return enable_driver(session, test, cmd, device_id)
+
+
+def unload_driver(session, vm, test, params, load_method='enable'):
+    """
+    Unload driver.
+
+    :param session: The guest session object
+    :param vm: vm object
+    :param test: kvm test object
+    :param params: the dict used for parameters
+    :param load_method: Load driver method
+    """
+    driver_name = params["driver_name"]
+    if load_method != 'enable':
+        devcon_path = params["devcon_path"]
+
+        device_name = driver_info_dict[driver_name]["device_name"]
+        device_hwid = driver_info_dict[driver_name]["hwid"]
+        return uninstall_driver(session, test, devcon_path, driver_name,
+                                device_name, device_hwid)
+    else:
+        cmd = params["driver_disable_cmd"]
```

Comment:
```suggestion
        cmd = '%s enable "%s"' % (params["devcon_path"], device_id)
```

Reply:
@vivianQizhu  many thanks, then it is more convenient.

run a case to confrim the update is oky first.

 (1/2) Host_RHEL.m9.u2.ovmf.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.rng_bat.q35: STARTED
 (1/2) Host_RHEL.m9.u2.ovmf.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.rng_bat.q35: PASS (124.08 s)
 (2/2) Host_RHEL.m9.u2.ovmf.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.rng_bat.iommu_enabled.q35: STARTED
 (2/2) Host_RHEL.m9.u2.ovmf.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.rng_bat.iommu_enabled.q35: PASS (115.81 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0